### PR TITLE
[PTBER072022] berndHavekost

### PR DESCRIPTION
**Pull request by hvkst**

### CODE REVIEW

File: src/functions-and-arrays.js [hunk 1]
```diff
@@ -103,17 +203,92 @@ const matrix = [
...
+  return createdMatrix;
+}
+
+function getRandomNumber(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min)) + min;
+}
 
+console.log(makeMatrix(10, 10, 1, 100));
```
Comment (trebeljahr) [inline src/functions-and-arrays.js:287]: all of the above bonus iterations is exceptional work! :partying_face: really well done. you deserve a tap on the shoulder or a trophy or something :)

**Review by trebeljahr — COMMENTED** [review]
think this might be the best version of this lab that I have ever seen. exceptional work.

File: src/functions-and-arrays.js [hunk 2]
```diff
@@ -1,86 +1,186 @@
...
-
-
+function uniquifyArray(array) {
+  // Elemente zu neuem Array pushen,
+  // oder, wenn schon vorhanden, skippen
+  let newArray = [];
+  if (!array.length) return null;
+  for (i of array) {
+    if (newArray.includes(i)) {
+      continue;
```
[the remark this answers is not in the document]
Comment (hvkst) [reply src/functions-and-arrays.js:130]: sure... nice!

File: src/functions-and-arrays.js (hunk 1 above)
Comment (hvkst) [reply src/functions-and-arrays.js:287]: Thank you very much for all your detailed comments! And the ["JS Black Magic"] :)

File: src/functions-and-arrays.js [hunk 3]
```diff
@@ -1,86 +1,186 @@
...
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring",
```
[the remark this answers is not in the document]
Comment (hvkst) [reply src/functions-and-arrays.js:120]: Is this about " " or ' ' ?

(Most of the code I see uses single quotes, but I´m so used to double quotes that it´s hard to change...)

File: src/functions-and-arrays.js [hunk 4]
```diff
@@ -1,86 +1,186 @@
...
+  return sum;
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(array) {
+  return !array.length ? null : array.reduce((a, b) => a + b) / array.length;
```
[the remark this answers is not in the document]
Comment (hvkst) [reply src/functions-and-arrays.js:60]: YES!

**Pull request closed without merging**